scalafix-core/shared/src/main/scala/scalafix/internal/rule/OrphanImplicits.scala

```diff
@@ -0,0 +1,66 @@
+package scalafix.internal.rule
+
+import scala.meta._
+import scalafix.internal.util.{ContextTraverser, SymbolOps}
+import scalafix.lint.LintCategory
+import scalafix.rule.RuleCtx
+import scalafix.util.SemanticdbIndex
+import scalafix.{LintMessage, Rule}
+
+final case class OrphanImplicits(index: SemanticdbIndex)
+    extends Rule("OrphanImplicits") {
+
+  override def description: String =
+    "Linter bans definitions of implicits F[G] unless they are on the companion of the F or the G"
```

Comment:
> Linter that reports an error on `implicit` instances of shape `F[G]` or `F` that do not belong in the companion objects of `F` or `G`

Reply:
this is definitely about things of shape `F[G]`, not about implicits of `F` on an `F`.

In fact I'd like a rule to ban implicits for anything except typeclasses :smile:

Reply:
(`ReaderT` is a far superior alternative to implicit configuration objects)

Reply:
The docs had an example of `implicit val f: Foo = null` so I assumed that was intended behavior too. Drop the "or `F`" part then